### ISSUE TITLE
Implement SweepResultAtom and collate functionality

### ISSUE THREAD

**Issue by dtch1997** [dtch1997/motools#171]
Add support for aggregating sweep results into a queryable atom with DataFrame export.

**Goal:**
Enable easy analysis of sweep results by creating a `SweepResultAtom` that tracks all workflows in a sweep and provides collation to DataFrame.

**Proposed implementation:**

1. **Create SweepResultAtom**
   - New atom type in `motools/atom/base.py`
   - Stores: list of workflow result IDs, sweep metadata (params, user, timestamp)
   - Method: `async def to_dataframe() -> pd.DataFrame`

2. **Update run_sweep()**
   - After workflows complete, create and save `SweepResultAtom`
   - Return both `list[WorkflowState]` and `SweepResultAtom` (or just the atom?)
   - Enable querying sweep results later

3. **Implement collation logic**
   - Extract config parameters from each workflow (e.g., n_epochs)
   - Extract eval metrics from EvalAtoms
   - Extract metadata: runtime, model_id, timestamps
   - Aggregate into DataFrame with columns:
     - Config params (sweep variables)
     - High-level experiment metadata
     - Metrics (task-dependent)

**Open questions:**

- **Task-dependent metrics:** Different eval tasks have different metrics (accuracy, f1, perplexity, etc.). How to specify which metric to extract?
  - Option A: Extract all metrics, let user filter
  - Option B: Allow per-task configuration of "primary metric"
  - Option C: Standardized metric naming convention

- **DataFrame schema:** What columns are essential?
  - Sweep params (e.g., `n_epochs`, `learning_rate`)
  - Model metadata (`model_id`, `base_model`)
  - Eval metrics (task-dependent)
  - Runtime info (`total_runtime_seconds`, `training_runtime`)
  - Timestamps (`started_at`, `completed_at`)

- **API design:** Should `run_sweep()` return `SweepResultAtom` or `list[WorkflowState]`?

**Use case:**
```python
# Run sweep
sweep_result = await run_sweep(...)

# Collate to DataFrame
df = await sweep_result.to_dataframe()
print(df[['n_epochs', 'accuracy', 'runtime']])

# Or load later
sweep_result = SweepResultAtom.load(sweep_id)
df = await sweep_result.to_dataframe()
```

**Related:**
- Enables experiment visualizer (#168)
- Builds on sweep functionality

**Note:** For MVP, bundle into `run_sweep()`. More sophisticated solution (separate save/load) can come later.
